packages/request-manager/src/controllerExternal.ts

```diff
@@ -0,0 +1,85 @@
+import { EventEmitter } from 'events';
+
+import { checkSocks5Proxy } from '@trezor/node-utils';
+
+import { waitUntil } from './utils';
+import { TOR_CONTROLLER_STATUS, TorControllerStatus, TorExternalConnectionOptions } from './types';
+
+const WAITING_TIME = 1_000;
+const MAX_TRIES_WAITING = 200;
+
+export class TorControllerExternal extends EventEmitter {
+    status: TorControllerStatus = TOR_CONTROLLER_STATUS.Stopped;
+    options: TorExternalConnectionOptions;
+
+    constructor(options: TorExternalConnectionOptions) {
+        super();
+        this.options = options;
+    }
+
+    private getIsStopped() {
+        return this.status === TOR_CONTROLLER_STATUS.Stopped;
+    }
+
+    private async getIsExternalTorRunning() {
+        let isSocks5ProxyPort = false;
+        try {
+            isSocks5ProxyPort = await checkSocks5Proxy(this.options.host, this.options.port);
+        } catch {
+            // Ignore errors.
+        }
+
+        return isSocks5ProxyPort;
+    }
+
+    private startBootstrap() {
+        this.status = TOR_CONTROLLER_STATUS.Bootstrapping;
+    }
+
+    private successfullyBootstrapped() {
+        this.status = TOR_CONTROLLER_STATUS.ExternalTorRunning;
+    }
+
+    public getTorConfiguration() {
+        return '';
+    }
+
+    public async waitUntilAlive() {
+        this.startBootstrap();
+        await waitUntil(
+            MAX_TRIES_WAITING,
+            WAITING_TIME,
+            async () => {
+                const isRunning = await this.getIsExternalTorRunning();
+                if (isRunning) {
+                    this.successfullyBootstrapped();
+                }
+
+                return isRunning;
+            },
+            () => {
+                return this.getIsStopped();
+            },
+        );
+    }
+
+    public async getStatus() {
+        const isExternalTorRunning = await this.getIsExternalTorRunning();
+
+        return new Promise(resolve => {
```

Comment:
This `new Promise` can be omitted, can't it?

Reply:
Also, I'm thinking whether `TorController` and `TorControllerExternal` can be merged/inherited or not, as they're pretty similar. But there are also differences, so I'm not sure yet.

Reply:
> This new Promise can be omitted, can't it?


I got rid of the unnecessary `new Promise`  https://github.com/trezor/trezor-suite/pull/15642/commits/b60c7e9a45c6c17c7f17e6b506c6aef493c6c0fd#diff-fa6e9c4a1d63d13830f1726b64551c8f1ca502b457fc2df80286010c4915692dR81


Regarding `TorController` and `TorControllerExternal` I was thinking a lot about it and initially started by having it all together. But found this approach better, more clear.